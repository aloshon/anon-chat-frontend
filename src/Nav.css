.navbar {
    background-color: #DDD;
    padding: 9px;
    text-align: center;
}

#navbar-logout {
    color: rgb(130, 130, 130);
}

.navbar-logout-div {
    overflow-wrap: break-word;
}

.navbar a {
    margin: 15px;
    color: rgb(130, 130, 130);
    display: inline-block;

}

.navbar a.active {
    color: rgb(183, 183, 183);
}

.navbar-brand {
    position: relative;
    left: 28px;
}

/* .navbar-items {
    font-size: 14px;
} */

/* .navbar a {
    margin: 10px 13px;
    color: rgb(183, 183, 183);
    display: inline-block;

} */