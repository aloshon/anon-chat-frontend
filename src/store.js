import  { composeWithDevTools } from "redux-devtools-extension";
import { persistStore, persistReducer } from "redux-persist";
import storage from "redux-persist/lib/storage";
import autoMergeLevel2 from "redux-persist/lib/stateReconciler/autoMergeLevel2";

import thunk from "redux-thunk";
import rootReducer from "./Reducers/rootReducer";
import { createStore, applyMiddleware } from "redux";

// only keep the darkMode reducer persistent in local storage
const persistConfig = {
  key: "root",
  whitelist:['darkMode'],
  storage,
  stateReconciler: autoMergeLevel2
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = createStore(
  persistedReducer,
  composeWithDevTools(
    applyMiddleware(thunk),
  )
);

export const persistedStore = persistStore(store);