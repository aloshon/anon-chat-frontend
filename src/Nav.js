import React, { useContext, useState } from "react";
import {NavLink} from "react-router-dom";
import "./Nav.css";
import Navbar from 'react-bootstrap/Navbar';
import { useSelector } from "react-redux";
import Nav from 'react-bootstrap/Nav';
import UserContext from "./UserContext";

// NavBar capitolization is crucial since Navbar and Nav are taken by Bootstrap **
/** 
 * NavBar renders different navbars depending if user is logged in or not
 * Grab user and logout function from UserContext,
 *  if user exists, return logged in navbar.
 *  If not then give navbar with sign up or log in
 */
const NavBar = () => {
    const {user, logout} = useContext(UserContext);
    const darkMode = useSelector(state => state.darkMode);
    const [expanded, setExpanded] = useState(false);

    if(user){
        return (
            <>
                <Navbar fixed="top" expanded={expanded} expand="md" bg={darkMode.nav} variant={darkMode.nav}>
                    <Navbar.Brand href="/" className="m-auto navbar-brand"><h2>anonChat</h2></Navbar.Brand>
                    <Navbar.Toggle onClick={() => setExpanded(expanded ? false : "expanded")} />
                    <Navbar.Collapse id="responsive-navbar-nav">
                        <Nav className="ml-auto navbar-items">
                            <NavLink onClick={() => setExpanded(false)} to="/create">Create New anonChat</NavLink>
                            <NavLink onClick={() => setExpanded(false)} to="/contacts">Contacts</NavLink>
                            <NavLink onClick={() => setExpanded(false)} to="/profile">Profile</NavLink>
                            <NavLink onClick={() => setExpanded(false)} to="/" id="navbar-logout">
                                <div className="navbar-logout-div" onClick={logout}>
                                    Logout
                                </div>
                            </NavLink>
                        </Nav>
                    </Navbar.Collapse>
                </Navbar>
            </>
        );
    }
    
    return (
            <Navbar fixed="top" bg={darkMode.nav} variant={darkMode.nav}>
                <Navbar.Brand href="/">anonChat</Navbar.Brand>
                <Nav className="ml-auto navbar-items">
                    <NavLink data-testid="signup" to="/signup">Sign Up</NavLink>
                    <NavLink to="/login">Log In</NavLink>
                </Nav>
            </Navbar>
    );

}

export default NavBar;