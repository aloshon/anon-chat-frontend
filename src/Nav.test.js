import { render, screen } from '@testing-library/react';
import Nav from './Nav';
import { Provider } from "react-redux";
import { store } from "./store"
import {BrowserRouter} from 'react-router-dom';
import UserContext from "./UserContext";


describe("Nav should render logged in version", () => {
    const user = {
        id: 1,
        username: "user1",
        isAdmin: false,
        blockList: []
    }

    function logout(user){
        user = undefined;
        return user
    }

    const {asFragment} = render(
      <Provider store={store}>
        <BrowserRouter>
            <UserContext.Provider value={{user, logout}}>
                <Nav/>
            </UserContext.Provider>
        </BrowserRouter>
      </Provider>
    );

    test("renders logged in version if user is defined", () => {
        const contacts = screen.getByText("Contacts");
        const profile = screen.getByText("Profile");
        expect(contacts).toBeInTheDocument();
        expect(profile).toBeInTheDocument();
    });
  
    test("matches snapshot", () => {
      expect(asFragment()).toMatchSnapshot();
    });
    
});
