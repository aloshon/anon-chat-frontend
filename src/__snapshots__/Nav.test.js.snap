// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should render logged in version matches snapshot 1`] = `<DocumentFragment />`;
